### ISSUE TITLE
[Feature Request] optionally allow restricting the region that locateOnScreen() searches for increased performance

### ISSUE THREAD

**Issue by pmcanseco** [asweigart/pyautogui#216]
Instead of searching the whole screen for each locateOnScreen() call, how about allowing an optional 4-tuple region parameter to make it so it only searches that region for a match, that way we don't waste time searching where we know we won't find anything?

**Comment by bjanzen** [comment]
Agreed.  I needed this, so I used Image.crop then loaded the cropped image to searched only that area for text.  But there's probably better/faster ways of doing this.

**Comment by GS-14** [comment]
Hello!

I went looking through the code to see how hard it would be to add a region parameter.
Turns out, according to this:
http://pyautogui.readthedocs.io/en/latest/screenshot.html?highlight=locate#the-locate-functions
there is no region parameter.

However, according to the source code:
https://github.com/asweigart/pyautogui/blob/master/pyautogui/screenshotUtil.py#L104
There is a region parameter. I haven't had a chance to test it out yet, but it seems like the feature is already there. It just seems to be missing from the documentation?

**Comment by bjanzen** [comment]
It is, and it's documented in pyscreeze.
